src/Symfony/Component/Security/Http/Tests/Firewall/SwitchUserListenerTest.php

```diff
@@ -118,7 +118,8 @@ public function testExitUserDoesNotRedirectToTargetRoute()
         $this->request->query->set('_switch_user', SwitchUserListener::EXIT_VALUE);
 
         $listener = new SwitchUserListener($this->tokenStorage, $this->userProvider, $this->userChecker, 'provider123', $this->accessDecisionManager, urlGenerator: $this->createMock(UrlGeneratorInterface::class), targetRoute: 'whatever');
-        $listener($this->event);
+        $this->assertTrue($listener->supports($this->event->getRequest()));
```

Comment:
this change is necessary because `supports()` sets the `_switch_user_username` attribute in the request: https://github.com/symfony/symfony/blob/7.4/src/Symfony/Component/Security/Http/Firewall/SwitchUserListener.php#L79

This is not new but I would like to challenge the `supports()` method manipulating the request.